extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -0,0 +1,491 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import axios, { AxiosResponse } from 'axios';
+import * as qs from 'qs';
+import * as url from 'url';
+
+import {
+	AzureAccountProviderMetadata,
+	Tenant,
+	AzureAccount,
+	Resource,
+	AzureAuthType,
+	Subscription
+} from '../interfaces';
+
+import { SimpleTokenCache } from '../simpleTokenCache';
+import { MemoryDatabase } from '../utils/memoryDatabase';
+const localize = nls.loadMessageBundle();
+
+export interface AccountKey {
+	/**
+	 * Account Key - uniquely identifies an account
+	 */
+	key: string
+}
+
+export interface AccessToken extends AccountKey {
+	/**
+	 * Access Token
+	 */
+	token: string;
+}
+
+export interface RefreshToken extends AccountKey {
+	/**
+	 * Refresh Token
+	 */
+	token: string;
+
+	/**
+	 * Account Key
+	 */
+	key: string
+}
+
+export interface TokenResponse {
+	[tenantId: string]: Token
+}
+
+export interface Token extends AccountKey {
+	/**
+	 * Access token
+	 */
+	token: string;
+
+	/**
+	 * TokenType
+	 */
+	tokenType: string;
+}
+
+export interface TokenClaims { // https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens
+	aud: string;
+	iss: string;
+	iat: number;
+	idp: string,
+	nbf: number;
+	exp: number;
+	c_hash: string;
+	at_hash: string;
+	aio: string;
+	preferred_username: string;
+	email: string;
+	name: string;
+	nonce: string;
+	oid: string;
+	roles: string[];
+	rh: string;
+	sub: string;
+	tid: string;
+	unique_name: string;
+	uti: string;
+	ver: string;
+}
+
+export type TokenRefreshResponse = { accessToken: AccessToken, refreshToken: RefreshToken, tokenClaims: TokenClaims };
+
+export abstract class AzureAuth {
+	protected readonly memdb = new MemoryDatabase();
+
+	protected readonly WorkSchoolAccountType: string = 'work_school';
+	protected readonly MicrosoftAccountType: string = 'microsoft';
```

Comment:
These feel like they should be defined in a more central location (or at least as const values somewhere)

Reply:
Yeah I'm not really sure where though